src/main/java/frc/team2412/robot/commands/shooter/ShooterTargetCommand.java

```diff
@@ -6,58 +6,41 @@
 
 import edu.wpi.first.wpilibj2.command.CommandBase;
 import frc.team2412.robot.subsystem.ShooterSubsystem;
-import frc.team2412.robot.subsystem.ShooterVisionSubsystem;
+import frc.team2412.robot.subsystem.TargetLocalizer;
 import frc.team2412.robot.util.ShooterDataDistancePoint;
 
 public class ShooterTargetCommand extends CommandBase {
     private final ShooterSubsystem shooter;
-    private final ShooterVisionSubsystem vision;
+    private final TargetLocalizer localizer;
     private final BooleanSupplier turretEnable;
 
-    public ShooterTargetCommand(ShooterSubsystem shooter, ShooterVisionSubsystem vision) {
-        this(shooter, vision, () -> false);
+    public ShooterTargetCommand(ShooterSubsystem shooter, TargetLocalizer localizer) {
+        this(shooter, localizer, () -> false);
     }
 
-    public ShooterTargetCommand(ShooterSubsystem shooter, ShooterVisionSubsystem vision, BooleanSupplier turretEnable) {
+    public ShooterTargetCommand(ShooterSubsystem shooter, TargetLocalizer localizer, BooleanSupplier turretButton) {
```

Comment:
I'd change `turretButton` since there's a [`Button`](https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/button/Button.html) class. I don't see a reason not to use `this.turretEnable = turretEnable`, especially since the other fields are set using `this.foo = foo`.

Reply:
booleansupplier more versatile

Reply:
I meant change the name of `turretButton`, I don't mind using `BooleanSupplier` at all.